src/sync/mod.rs

```diff
@@ -30,11 +30,21 @@ pub enum ProtocolResponse {
     Failure,
 }
 
+#[derive(Debug)]
+/// The state of the Convex WebSocket connection
+pub enum WebSocketState {
+    /// The WebSocket is open and connected
+    Connected,
+    /// The WebSocket is closed and reconnecting
+    Reconnecting,
```

Comment:
Is this also the initial state before the socket has connected? Would it be better to just call it `Connecting` and let the thing consuming the state figure out if it's a reconnect or initial connect?